extruct/rdfa.py

```diff
@@ -28,6 +29,104 @@
 
 
 class RDFaExtractor(object):
+    
+    # expands namespace to match with returned json (ex: og -> 'http://ogp.me/ns#')
+    def replaceNS(self, prop, html_element, head_element):
+        
+        # context namespaces taken from pyrdfa3
+        # https://github.com/RDFLib/PyRDFa/blob/master/pyRdfa/initialcontext.py
+        context = {
+            'owl'   : 'http://www.w3.org/2002/07/owl#',
+            'gr'    : 'http://purl.org/goodrelations/v1#',
+            'ctag'    : 'http://commontag.org/ns#',
+            'cc'    : 'http://creativecommons.org/ns#',
+            'grddl'   : 'http://www.w3.org/2003/g/data-view#',
+            'rif'   : 'http://www.w3.org/2007/rif#',
+            'sioc'    : 'http://rdfs.org/sioc/ns#',
+            'skos'    : 'http://www.w3.org/2004/02/skos/core#',
+            'xml'   : 'http://www.w3.org/XML/1998/namespace',
+            'rdfs'    : 'http://www.w3.org/2000/01/rdf-schema#',
+            'rev'   : 'http://purl.org/stuff/rev#',
+            'rdfa'    : 'http://www.w3.org/ns/rdfa#',
+            'dc'    : 'http://purl.org/dc/terms/',
+            'foaf'    : 'http://xmlns.com/foaf/0.1/',
+            'void'    : 'http://rdfs.org/ns/void#',
+            'ical'    : 'http://www.w3.org/2002/12/cal/icaltzd#',
+            'vcard'   : 'http://www.w3.org/2006/vcard/ns#',
+            'wdrs'    : 'http://www.w3.org/2007/05/powder-s#',
+            'og'    : 'http://ogp.me/ns#',
+            'wdr'   : 'http://www.w3.org/2007/05/powder#',
+            'rdf'   : 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
+            'xhv'   : 'http://www.w3.org/1999/xhtml/vocab#',
+            'xsd'   : 'http://www.w3.org/2001/XMLSchema#',
+            'v'     : 'http://rdf.data-vocabulary.org/#',
+            'skosxl'  : 'http://www.w3.org/2008/05/skos-xl#',
+            'schema'  : 'http://schema.org/',
+        }
+
+        if 'http://' in prop:
+            return prop
+
+        prefix = prop.split(':')[0]
+        
+        match = None
+        if head_element.get('prefix'):
+            match = re.search(prefix + ': [^\s]+', head_element.get('prefix'))
+
+        # if namespace taken from prefix attribute in head tag
+        if match:
+            ns = match.group().split(': ')[1]
+            return ns + prop.split(':')[1]
```

Comment:
If prop is wrong (e.g. "baddefinedprop") this line would fail (index out of range).

Reply:
The if condition will only pass if the property is well defined - in this case only if the namespace was in the prefix attribute of the head tag. I tried with a wrong prop ("baddefinedprop"), and it worked fine.

Reply:
I see, good point. I missed that.

Reply:
Oh right, I wouldn't fail for "baddefinedprop". But it would fail for "owl" property if head element is something like:
```html
<head prefix="owl:http://www.w3.org/2002/07/owl#">
```

Reply:
I think this is now solved.